### PR TITLE
DAT-2845 add infobox parser api

### PR DESCRIPTION
**Pull request by idradm**

### CODE REVIEW

File: extensions/wikia/PortableInfobox/PortableInfobox.setup.php [hunk 1]
```diff
@@ -54,3 +55,6 @@
 
 // i18n mapping
 $wgExtensionMessagesFiles[ 'PortableInfobox' ] = $dir . 'PortableInfobox.i18n.php';
+
+//MW API
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/PortableInfobox.setup.php]: nitpick: a space :)

File: extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php [hunk 2]
```diff
@@ -43,6 +43,31 @@ public static function replaceInfoboxMarkers( &$parser, &$text ) {
...
+	 * @param $markup
+	 * @param Parser $parser
+	 * @param PPFrame $frame
+	 * @return array|string
+	 * @throws UnimplementedNodeException when node used in markup does not exists
+	 * @throws XmlMarkupParseErrorException xml not well formatted
+	 */
+	public function render( $markup, Parser $parser, PPFrame $frame, $params = null ) {
+		$infoboxParser = new Wikia\PortableInfobox\Parser\XmlParser(
+			array_merge( $frame->getNamedArguments(), $frame->getArguments() ) );
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php]: Please check the return value type, if it is null we should handle it prior to merging

Comment (inez) [inline extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php]: Well - I like how some extra clean up came out of this (separating out method "render")

File: extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php [hunk 3]
```diff
@@ -43,6 +43,30 @@ public static function replaceInfoboxMarkers( &$parser, &$text ) {
...
 	/**
+	 * @param $markup
+	 * @param Parser $parser
+	 * @param PPFrame $frame
+	 * @return string
+	 * @throws UnimplementedNodeException when node used in markup does not exists
+	 * @throws XmlMarkupParseErrorException xml not well formatted
+	 */
+	public function render( $markup, Parser $parser, PPFrame $frame, $params = null ) {
+		$infoboxParser = new Wikia\PortableInfobox\Parser\XmlParser( $this->getFrameParams( $frame ) );
```
Comment (SebastianMarzjan) [inline extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php]: :+1: for extracting this method. Reads much easier now at this level of abstraction

**Comment by SebastianMarzjan** [comment]
LGTM

**Pull request merged**